osfoffline/polling_osf_manager/polling.py

```diff
@@ -371,25 +371,23 @@ def _check_file_folder(self,
         else:
             raise ValueError('in some weird state. figure it out.')
 
-        assert local_file_folder is not None
-        assert remote_file_folder is not None
```

Comment:
The assertions prevented this function from *ever* being used if the preconditions were not met, whereas nested things inside an if block means that a failure will pass silently.

I'm not a fan of assertions for general checks (like `try: raise AssertionError  / except AssertionError as e`, but in this case they provide a very strict sanity check. I'm not convinced that the if statements (which fail silently if the condition is not met) are a complete replacement.